### PR TITLE
remove reference to blueprint.js in css

### PR DESCRIPTION
**Pull request by echarles**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/12331

## Code changes

Some CSS rules aimed to override blueprint.js are removed in ui-components.

## User-facing changes

Let's see that...

### CODE REVIEW

**Comment by fcollonval** [comment]
I'm :+1: for this one - especially to remove those `unset` rules (on `*` :scream:)

But that will definitely impact downstream extensions (in particular the box-sizing rule).

**Comment by echarles** [comment]
I have been impacted by the box-sizing rule, which has driven me to look at all those overrides. Surprisingly the UI remains ok at first sight, and the regression ui reports the following issues.

```
  26 failed
    [galata] › test/galata/notebook.spec.ts:92:3 › Notebook Tests › Open and run cell by cell ======
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:38:3 › Collapsible Headings; showHCB › Collapse Heading; showHCB 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:129:3 › Collapsible Headings; keyboard navigation › Collapse Previous Header 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:138:3 › Collapsible Headings; keyboard navigation › Collapse Previous Headers 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:149:3 › Collapsible Headings; keyboard navigation › ReExpand Headers 01 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:165:3 › Collapsible Headings; keyboard navigation › ReExpand Headers 02 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:178:3 › Collapsible Headings; keyboard navigation › ReExpand Headers 03 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:198:3 › Collapsible Headings; keyboard navigation › Add Header Below 01 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:213:3 › Collapsible Headings; keyboard navigation › Add Header Below 02 
    [jupyterlab] › test/jupyterlab/collapsible-headings.test.ts:229:3 › Collapsible Headings; keyboard navigation › Add Header Below 03 
    [jupyterlab] › test/jupyterlab/debugger.test.ts:34:3 › Debugger Tests › Start debug session ====
    [jupyterlab] › test/jupyterlab/debugger.test.ts:69:3 › Debugger Tests › Start debug session (Script) 
    [jupyterlab] › test/jupyterlab/general.test.ts:8:3 › General Tests › Launch Screen =============
    [jupyterlab] › test/jupyterlab/general.test.ts:13:3 › General Tests › Enter Simple Mode ========
    [jupyterlab] › test/jupyterlab/general.test.ts:28:3 › General Tests › Toggle Dark theme ========
    [jupyterlab] › test/jupyterlab/notebook-run.test.ts:34:3 › Notebook Run › Run Notebook and capture cell outputs 
    [jupyterlab] › test/jupyterlab/sidebars.test.ts:17:5 › Sidebars › Open Sidebar tab filebrowser =
    [jupyterlab] › test/jupyterlab/sidebars.test.ts:17:5 › Sidebars › Open Sidebar tab jp-property-inspector 
    [jupyterlab] › test/jupyterlab/sidebars.test.ts:17:5 › Sidebars › Open Sidebar tab jp-running-sessions 
    [jupyterlab] › test/jupyterlab/sidebars.test.ts:17:5 › Sidebars › Open Sidebar tab extensionmanager.main-view 
    [jupyterlab] › test/jupyterlab/sidebars.test.ts:60:3 › Sidebars › Capture File Browser on right 
    [jupyterlab] › test/jupyterlab/toc-running.test.ts:23:3 › ToC Running indicator › should display running indicators 
    [jupyterlab] › test/jupyterlab/toc-running.test.ts:37:3 › ToC Running indicator › should display running indicator on first visible top level 
    [jupyterlab] › test/jupyterlab/toc.test.ts:39:3 › Table of Contents › Add tags =================
    [jupyterlab] › test/jupyterlab/workspace.test.ts:42:3 › Workspace › should reset the workspace =
    [jupyterlab] › test/jupyterlab/workspace.test.ts:93:3 › Workspace › should reset the workspace in simple mode 
```

**Comment by echarles** [comment]
This is a diff example. The rest of the diffs are in the same vein. Should I just trigger a snapshot update via `please update galata snapshots`?

cddc01d866e27f7e3690b4f33a76c93e71620f84

**Comment by fcollonval** [comment]
To update both UI tests and UI documentation snapshots, you can call directly `please update snapshots`

**Comment by echarles** [comment]
please update snapshots

**Comment by echarles** [comment]
@fcollonval the above `please update snapshots` looks to have no effect.

**Comment by fcollonval** [comment]
The bot failed to push the new snapshots:

> https://github.com/jupyterlab/jupyterlab/runs/5828286417

**Comment by echarles** [comment]
please update snapshots

**Comment by fcollonval** [comment]
@echarles is your repo/branch protected??
The error changed - but the push is still blocked:

```
To https://github.com/jupyterlab/jupyterlab
 ! [remote rejected]       feat/remove-css-blueprint-ref -> refs/pull/12332/head (deny updating a hidden ref)
error: failed to push some refs to 'https://github.com/jupyterlab/jupyterlab'
```

What I don't get is why is it targeting the reference repository...

**Comment by echarles** [comment]
@fcollonval Oops I think I hit the same issue today on another PR. Forking in a org and opening a PR has some limitations...

**Comment by echarles** [comment]
> What I don't get is why is it targeting the reference repository...

@fcollonval  Does it work on a PR you have opened atm?

File: packages/ui-components/style/base.css [hunk 1]
```diff
@@ -19,41 +19,6 @@
...
-
-*,
-*::before,
-*::after {
-  box-sizing: unset;
-}
-
-body {
-  color: var(--jp-ui-font-color1);
-  font-size: var(--jp-ui-font-size1);
```
Comment (jasongrout) [inline packages/ui-components/style/base.css:35]: These two rules look like possibly important rules to keep, unless they are set elsewhere? Other than that, I'm so glad to see these go!

Comment (echarles) [reply packages/ui-components/style/base.css]: I have added them back. Thx @jasongrout

**Comment by fcollonval** [comment]
> What I don't get is why is it targeting the reference repository...

It seems it is always like that so ignoring that comment

> See my comments on https://github.com/ipython/ipykernel/pull/908#issuecomment-1088492275

Ok so I'll look at the action to save the snapshots

> Does it work on a PR you have opened atm?

Not at the moment but I did not see any errors for a couple of users in the last 2 weeks.

**Comment by fcollonval** [comment]
please update snapshots

**Comment by fcollonval** [comment]
@echarles you will be able to get the new snapshots as artifacts of https://github.com/jupyterlab/jupyterlab/actions/runs/2141009732

**Comment by echarles** [comment]
@fcollonval My last commit has the snapshot updated. I can see things to fix. The work is just starting to get the correct CSS tunings.

**Comment by echarles** [comment]
For example

Untitled4

**Comment by fcollonval** [comment]
I would say that the vertical compression is a good consequence - it does not hurt readability and allows more information on the screen. The only bug I'm seeing is for the launcher cards where the labels can be partially hidden.

**Comment by echarles** [comment]
> I would say that the vertical compression is a good consequence - it does not hurt readability and allows more information on the screen.

+1 

> The only bug I'm seeing is for the launcher cards where the labels can be partially hidden.

Yes, I have seen that also. I can push the last mile on that assuming there is n willingness to get it merged once ready. That change is not huge in terms of code, but could have unexpected side-effects on extensions...

BTW does the `please update snapshots` is supposed to work for me?

**Comment by fcollonval** [comment]
> Yes, I have seen that also. I can push the last mile on that assuming there is n willingness to get it merged once ready. That change is not huge in terms of code, but could have unexpected side-effects on extensions...

I think for 4.0 this is fine.

> BTW Is the `please update snapshots` supposed to work for me?

I cannot find the reference any more - but I remember you won't be able to do so because your fork is in an organization. But the update snapshots are now uploaded as artifacts in the job triggered by `please update snapshots`. So you can push a new commit manually using those snapshots.

**Pull request closed without merging**